File: internal/module/user/user_entity.go
package user

import (
	"time"
)

// User entity
type User struct {
	ID        uint64    `gorm:"primary_key;auto_increment" json:"id"`
	Email     string    `gorm:"type:varchar(256);unique;unique_index;not null" json:"email"`
	Password  string    `gorm:"type:varchar(512)" json:"password"`
	MaxTodo   int       `gorm:"type:integer" json:"max_todo"`
	CreatedAt time.Time `gorm:"default:CURRENT_TIMESTAMP" json:"created_at"`
	UpdatedAt time.Time `gorm:"default:CURRENT_TIMESTAMP" json:"updated_at"`
}

// ToObject func
func (user *User) ToObject() map[string]interface{} {
	return map[string]interface{}{
		"id":        user.ID,
		"email":     user.Email,
		"maxToto":   user.MaxTodo,
		"createdAt": user.CreatedAt,
		"updatedAt": user.UpdatedAt,
	}
}

File: togo-js/src/models/user.entity.ts
import {
  Entity,
  BaseEntity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  OneToMany,
  Index,
  DeleteDateColumn,
} from 'typeorm';
import { Task } from './task.entity';

@Entity()
export class User extends BaseEntity {
  @PrimaryGeneratedColumn('increment')
  id: number;

  @Index({ unique: true })
  @Column({ type: 'varchar' })
  userId: string;

  @Column({ type: 'varchar', select: false })
  password: string;

  @Column({ type: 'int4', default: 5 })
  maxTodo: number;

  @OneToMany(() => Task, (task) => task.user)
  tasks: Task[];

  @CreateDateColumn({ type: 'timestamptz' })
  createdAt: Date;

  @UpdateDateColumn({ type: 'timestamptz', nullable: true })
  updatedAt: Date;
}
